ex-3-recursion/look-for-key.js

```diff
@@ -0,0 +1,21 @@
+function lookForKey( arr ){
+
+    for (let i = 0; i < arr.length; i++) {
+        if( arr[i] == "key" ){
+            console.log( "Key found.");
+            //return "Key found.";   - how to return when found?
+        } else if( Array.isArray( arr[i] ) ){
+            lookForKey( arr[i] );
+        }
+    }
+}
```

Comment:
Sweet! How "deep" is your algorithm able to go?

Reply:
It's checking recursively, so it should continue until the key is found or until some recursion limit is hit(I've just now learned that might be a thing).  But I think this is probably a terrible implementation.  I'm able to log when the key is found, but can't return.  I know that's because it's trying to return inside the recursive calls, but I don't fully understand how to make the function return when found.

Reply:
> it should continue until the key is found or until some recursion limit is hit

Yup!

> But I think this is probably a terrible implementation.

You're not that far off! Compare what you did with quicksort here: https://github.com/mdmoore/grokking-algorithms/pull/6/files/bb445d6c9f2e4e144cac0398e185869aaa2c9150#diff-575104ee552503eb8dd8fc43e901ee6063941e28703eaa05b45c35e758283d4b

It may help to get it working with the following two examples in order:

```
["key"]

[["key"]]
```

Get it working with the simplest example `["key"]` first, then see what you need to change to make it work for the next simplest case `[["key"]]`.